libvirt.yaml

```diff
@@ -0,0 +1,266 @@
+  agent_user:
+    default: { get_secret: agent_user }
+
+  disk_size:
+    default: 24GB
+
+  # cluster(kvm) settings
+  cluster_user:
+    type: string
+    description: >
+      Cluster user name.
+    default: { get_secret: libvirt_cluster_user }
+
+  cluster_key:
+    type: string
+    description: >
+      SSH key for cluster user.
+    default: { get_secret: libvirt_cluster_key }
+
+  manager_network:
+    type: string
+    description: >
+      Manager network.
+    default: { get_secret: libvirt_common_network }
+
+  cluster_host:
+    type: string
+    description: >
+      LibVirt computer ip address.
+    default: { get_secret: libvirt_cluster_host }
+
+  external_ip:
+    type: string
+    description: >
+      List external ip's separated by comma.
+    default: { get_secret: libvirt_cluster_external_ip }
+
+  external_dev:
+    type: string
+    description: >
+      Device with external access (same device as used for cluster_host ip)
+    default: { get_secret: libvirt_cluster_external_dev }
+
+  cpus:
+    type: integer
+    default: 2
+
+  memory:
+    type: integer
+    description: RAM in MB
+    default: 4096
+
+  memory_size:
+    type: integer
+    description: RAM in KB (real size)
+    default: 4194304
+
+relationships:
+
+  vm_connected_to_storage:
+    derived_from: cloudify.relationships.contained_in
+    target_interfaces:
+      cloudify.interfaces.relationship_lifecycle:
+        preconfigure:
+          implementation: scripts/libvirt/vm_preconfigure.py
+          executor: central_deployment_agent
+          inputs: {}
+
+node_types:
+
+  kubernetes_host:
+    derived_from: cloudify.nodes.Compute
+    properties:
+      kubecpu:
+        # by default: amount of cpu's on vm avaible for kubepods,
+        # in most cases it must be same as full count of cpu on vm
+        default: { get_input: cpus }
+      kubemem:
+        # by default: amount of memory on vm avaible for kubepods,
+        # in most cases it must be same as full count of memory on vm
+        default: { get_input: memory }
+      use_public_ip:
+        default: "private"
+      agent_config:
+        default:
+          user: { get_secret: agent_user }
+          key: { get_secret: agent_key_private }
+          install_method: remote
+
+node_templates:
+
+  vm_download:
+    type: cloudify.nodes.WebServer
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: fabric.fabric_plugin.tasks.run_script
+          inputs:
+            fabric_env: &fab_env
+              user: { get_input: cluster_user }
+              key_filename: { get_input: cluster_key }
+              host_string: { get_input: cluster_host }
+            script_path: scripts/libvirt/download_vm.sh
+            CATALOG_URL: https://cloud.centos.org/centos/7/images/
+            DISK: CentOS-7-x86_64-GenericCloud-1805.qcow2
+
+  common_network:
+    type: cloudify.libvirt.network
+    properties:
+      libvirt_auth: &libvirt_auth
```

Comment:
To be consistent with other blueprints, let's use the `dsl_definitions` mechanism for authorization.